Improve Interactive CLI documentation (#20013)
ydb-platform/ydb: ydb/docs/en/core/reference/ydb-cli/interactive-cli.md

> * `global options` — [global parameters](commands/global-options.md).

Reviewer: тут нужен самый общий пример, до отдельных фичей

## Features {#features}

> * [Query history](#query-history).

Reviewer: 1. Точки надо поубирать из всех пунктов меню
2. Пункты меню я бы переставила в след. порядке:
 - подсветка синтаксиса
 - горячие клавиши
 - история запросов
 - спец команды
 - автодополнение

* Syntax highlighting based on YQL grammar.
* [Hotkeys](#hotkeys).
* [Special commands](#spec-commands).
* [Auto completion](#auto-completion).

### Query history {#query-history}

> Using up and down arrow keys you can navigate through the query history:

Reviewer: ```suggestion
You can navigate through the query history using up and down arrow keys:
```

![History](_assets/history.gif)

History is stored locally and persists between CLI launches.

> A query search function (`CTRL + R`) is also supported:

Reviewer: а на макбуке будет Cmd+R, или тоже Ctrl+R ?

Author: тоже Ctrl+R

![Search](_assets/history-search.gif)

## Hotkeys {#hotkeys}

| Hotkey        | Description                                                               |
|---------------|---------------------------------------------------------------------------|

> | `CTRL + D`    | Exits interactive mode.                                                   |

Reviewer: эту строку в конец таблицы

| `Up arrow`    | Shows previous query from history.                                        |
| `Down arrow`  | Shows next query from history.                                            |
| `TAB`         | Completes the current word based on YQL syntax.                          |
| `CTRL + R`    | Searches for a query in history containing a specified substring.         |

## Special commands {#spec-commands}

> Special commands are CLI-specific commands and are not part of the YQL syntax. They are intended for performing various functions that cannot be accomplished through a YQL query.

Reviewer: ```suggestion
Special commands are CLI-specific commands and are not part of the YQL syntax. They are developed for performing various functions that cannot be accomplished through a YQL query.
```

* `global options` — [global parameters](commands/global-options.md).

Example usage:

![Example](_assets/general-example.gif)

## Features {#features}

* [Syntax highlighting](#syntax-highlighting)
* [Hotkeys](#hotkeys)
* [Query history](#query-history)
* [Auto completion](#auto-completion)
* [Special commands](#spec-commands)

> ### Syntax highlighting {#syntax-highlighting}

Reviewer: ```suggestion
## Syntax highlighting {#syntax-highlighting}
```

* `global options` — [global parameters](commands/global-options.md).

Example usage:

![Example](_assets/general-example.gif)

The interactive query execution mode in {{ ydb-short-name }} CLI offers the following features:

* [Syntax highlighting](#syntax-highlighting)
* [Hotkeys](#hotkeys)
* [Query history](#query-history)
* [Auto completion](#auto-completion)
* [Special commands](#spec-commands)

## Syntax highlighting {#syntax-highlighting}

![Syntax highlighting](_assets/highlighting.jpg)

Interactive mode supports YQL syntax highlighting, which helps to better understand query structure. Different colors are used for the following groups of elements:

* YQL keywords (SELECT, FROM, WHERE, INSERT, UPDATE and others)
* Table and column names
* String literals (text in quotes)
* Numeric literals
* Operators (=, <, >, +, - and others)
* Special characters (brackets, commas, dots)

> * Comments

Reviewer 2: Why? If they change, you'll be able to just update it.

| `CTRL + R`    | Searches for a query in history containing a specified substring.         |
| `CTRL + D`    | Exits interactive mode.                                                   |

## Query history {#query-history}

You can navigate through the query history using up and down arrow keys:

![History](_assets/history.gif)

History is stored locally and persists between CLI launches.

A query search function (`CTRL + R`) is also supported:

![Search](_assets/history-search.gif)

## Auto completion {#auto-completion}

Auto completion helps you write queries more efficiently. While typing, it suggests possible completions for the current word based on YQL syntax.

It also searches for schema object names in database where possible.

There are two types of suggestions: auto completion by pressing `TAB` key and interactive hints.

### Auto completion by pressing `TAB` key {#auto-completion-tab}

While in interactive mode, pressing `TAB` key shows a list of suggestions of completion of current word according to the YQL syntax.

![Auto completion](_assets/candidates.gif)

If there is only one available option, pressing TAB will automatically complete the current word.

> If all available options share a common prefix, pressing TAB will automatically insert it.

Reviewer 2: How to choose the option you like if there are multiple of them?

Author: Only by continuing typing. These options are just the information to consider

> If all available options share a common prefix, pressing TAB will automatically insert it.

Reviewer 2: How to choose the option you like if there are multiple of them?

> ### Interactive Hints {#interactive-hints}

Reviewer 2: The documentation is currently moving from `## Regular case` headers to `## Title Case` headers like this, but at least each file should be consistent in this regard.

> While typing in interactive mode, a list of hints will appear under the cursor, showing first 4 suggestions of completion of current word according to the YQL grammar.

Reviewer 2: It doesn't seem to follow the grammar accurately. For example, afaik, here's only `INSERT` is valid for YDB:
![Screenshot 2568-07-01 at 10 46 30](https://github.com/user-attachments/assets/2d01c942-ddb7-4a4c-bfd9-efa262e2801c)

Reviewer 3: @pnv1, can you also, please, share a revision of the code, that screenshots was taken. If it is a fresh main, then it seems to be a bug, because `import_stmt` is expected to be ignored.

Reviewer 3: I propose for now to use such an input for a screenshot to hide this issue. Then a fix will be made.

> ### Auto completion by pressing `TAB` key {#auto-completion-tab}
> 
> While in interactive mode, pressing `TAB` key shows a list of suggestions of completion of current word according to the YQL syntax.
> 
> ![Auto completion](_assets/candidates.gif)
> 
> If there is only one available option, pressing TAB will automatically complete the current word.
> If all available options share a common prefix, pressing TAB will automatically insert it.
> 
> ### Interactive Hints {#interactive-hints}
> 
> While typing in interactive mode, a list of hints will appear under the cursor, showing first 4 suggestions of completion of current word according to the YQL grammar.
> 
> ![Interactive hints](_assets/hints.gif)
> 
> This feature provides quick guidance without overwhelming you with all possible options, helping you stay on track while writing queries.

Reviewer 2: ```suggestion
### Auto completion by pressing the `TAB` key {#auto-completion-tab}

While in interactive mode, pressing the `TAB` key shows a list of suggestions for completing the current word according to the YQL syntax.

![Auto completion](_assets/candidates.gif)

If there is only one available option, pressing `TAB` will automatically complete the current word.
If all available options share a common prefix, pressing `TAB` will automatically insert it.

### Interactive hints {#interactive-hints}

While typing in interactive mode, a list of hints appears under the cursor, showing the first four suggestions for completing the current word according to the YQL grammar.

![Interactive hints](_assets/hints.gif)

This feature provides quick guidance without overwhelming you with all possible options, helping you stay on track while writing queries.
```

* `global options` — [global parameters](commands/global-options.md).

{% note info %}

> Note that the command still needs [connection parameters](./connect.md) to be set. It may be a default or explicitly specified profile and/or a set of connection parameters.

Reviewer 2: ```suggestion
Note that the command still requires [connection parameters](./connect.md) to be set. You can supply them by having the default profile, via an explicitly specified profile, or by passing a set of connection parameters.
```

         TotalDurationMs: 0
         TotalOutputRows: 1
```

## Auto completion {#auto-completion}

Auto completion helps you write queries more efficiently. While typing, it suggests possible completions for the current word based on YQL syntax.

It also searches for schema object names in database where possible.

There are two types of suggestions: auto completion by pressing `TAB` key and interactive hints.

### Auto completion by pressing `TAB` key {#auto-completion-tab}

> While in interactive mode, pressing `TAB` key will show a list of suggestions of completion of current word according to the YQL syntax.

Reviewer: не описано, как выбрать из предложенного списка

![Auto completion](_assets/candidates.gif)

If there is only one suggestion available, pressing TAB will complete current word to it.

### Interactive Hints {#interactive-hints}

> While typing in interactive mode, a list of hints will appear under the cursor, showing first 4 suggestions of completion of current word according to the YQL grammar.

Reviewer: не описано, как выбрать из предложенного списка (и можно ли)

Author: Нельзя
